### ISSUE TITLE
request for mac arm agent

### ISSUE THREAD

**Issue by Raxiel1987** [Ylianst/MeshCentral#6308]
i see the agent for the x86-64 bit mac os operating system, but not for arm
i see the binary installer for arm too

but the agent will be available ?

**Comment by si458** [comment]
The mpkg is currently universal, so it should work for both arm/x64 if I remember
However, you can already download the binary for universal/x64/arm

**Comment by si458** [comment]
@Raxiel1987 ah right ok not a problem!
the binarys etc are all there already!
we just need to create the steps/scripts to use either x64 or arm for the mpkg!
so more options to be added!
mpkg universal
mpkg arm
mpkg x64

**Comment by si458** [comment]
@Raxiel1987 its actually only a few script changes! leave it with me, ill have a look now if i can

**Comment by si458** [comment]
@Raxiel1987 no select the macos option, not the binary option! First drop down with windows/linux/etc...

**Comment by si458** [comment]
so long as u can install correctly, its a web ui fix 👍

**Comment by si458** [comment]
was it defo the `id=29` you downloaded?

if you download the mpkg, then `right click` and `show package contents`
then into `Contents` and `Packages`
then again `right click` the `internal.pkg` and `show package contents`
then into `Contents`
then open up a terminal shell and type `file ` then drag the `meshagent_osx64.bin` to the terminal and press enter
you should get an option saying what type of file it is?
output should be something like mine
```
simon@SiMacBookPro16 ~ % file /Users/simon/Downloads/meshagent.mpkg/Contents/Packages/internal.pkg/Contents/meshagent_osx64.bin
/Users/simon/Downloads/meshagent.mpkg/Contents/Packages/internal.pkg/Contents/meshagent_osx64.bin: Mach-O 64-bit executable arm64
```

**Comment by si458** [comment]
i have also just added the PR to show the extra mpkgs in the web ui https://github.com/Ylianst/MeshCentral/commit/08877844c4040b8323c3e83ccb8fd81dbcd1b765
1. BACKUP YOUR MACHINE/FILES! (im not responsible if if breaks and you cant roll back!)
2. stop meshcentral
3. install the test meshcentral depending on how you installed meshcentral to begin with
`npm install Ylianst/MeshCentral` - if you used a normal `npm` install/setup (note this will lose translations)
`ghcr.io/ylianst/meshcentral:master` - the master docker image for testing
4. start meshcentral

**Comment by si458** [comment]
ive found the bug and its a simple fix!
you can test it yourself
download the mpkg for arm
`show contents` then contents then edit the `distribution.dist` file
then add `<options hostArchitectures="arm64,x86_64" />` right above the last line `</installer-script>` 
then save, close, and reopen the mpkg!
its because mac defaults all pkg and mpkg to x86-64, unless u tell it that it supports arm
so hense it was ALWAYS saying it needed rosetta even tho it didnt!

**Comment by si458** [comment]
all fixed now! 61f1c22

so wait 10-15mins, then do the steps above again https://github.com/Ylianst/MeshCentral/issues/6308#issuecomment-2271486645

**Comment by si458** [comment]
yipee! great will close the issue then 👍 
p.s. dont forget to donate to help keep the project alive ❤️ https://www.si458.co.uk/2024/01/05/donation/
